### DISCUSSION TITLE
What's the best way to let a Hugging Face Space pull a private Python library?

### DISCUSSION THREAD

**Discussion by umarbutler** [community/community#145475] [Other Feature Feedback, Questions, & Ideas]
Say you have a Hugging Face Space that depends on a Python library that you host on GitHub in a private repository. Normally, you'd simply add `git+https://github.com/umarbutler/mylibrary.git` to your `requirements.txt` and then Hugging Face would automatically install the library on startup.

If the repository is private, however, you need to authenticate to be able to pull the repository. So then you could do `git+https://${GIT_TOKEN}@github.com/umarbutler/mylibrary.git` and then add `GIT_TOKEN` as a secret in your space, GIT_TOKEN being a personal access token. The problem is when you are trying to do this for an organisation and there is no guarantee that the employee who used their personal access token will always be there. What do you do in that case?

**Comment by kivanc57** [comment]
You can indeed limit users' access with different policies in the organisational level, you just have to be `admin` on the repository. The place where you do is the same as where you usually create the secrets.

For more info, check these links:
https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/managing-development-environment-secrets-for-your-repository-or-organization#adding-secrets-for-an-organization 
https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/managing-development-environment-secrets-for-your-repository-or-organization#reviewing-access-to-organization-level-secrets
